main.cpp

```diff
@@ -115,17 +126,21 @@ struct <#structname2#>
 
 int main()
 {
-    T <#name1#>( , );                                             //6
-    T <#name2#>( , );                                             //6
+    SoccerBall walmartBall( 4.4f, "Wilson");                                             //6
+    SoccerBall targetBall(5.5f, "Nike");                                             //6
     
-    <#structName1#> f;                                            //7
-    auto* smaller = f.compare( , );                              //8
+    SoccerBallSize f;                                            //7
+    auto* smaller = f.compare(&walmartBall, &targetBall);                              //8
+    if(smaller == nullptr)
+        return -1;
     std::cout << "the smaller one is << " << smaller->name << std::endl; //9
```

Comment:
Don't return prematurely. 
Instead, print out the statement if `smaller` is NOT a nullptr
- [ ] revise